### PR TITLE
MEN-505: Fix - Now throws an error when committing nothing.

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: Title

Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

File: device.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type device struct {
 	BootEnvReadWriter
 	Commander
 	*partitions
+	hasUpdateCb func(d *device) (bool, error)
```
Comment (bboozzoo) [inline device.go]: It looks to me that this callback is only needed for testability sake, in which case it makes no sense to have it here and instead update the tests to inject proper data as `BootEnvReadWriter` interface.

Comment (oleorhagen) [inline device.go]: @pasinskim

Comment (bboozzoo) [inline device.go]: An example of injecting `BootEnvReadWriter` with desired data is right here: https://github.com/mendersoftware/mender/pull/218/commits/f3e8bea7137b061c08ebc6c95cf30c0dba94e499#diff-eed9e366efcb961973c3dbd3d4b7d620R172

Comment (pasinskim) [inline device.go]: @bboozzoo the problem is that we need to read the env twice, and there is no way to mock env read/writer to return two different things for two different calls.

Comment (bboozzoo) [inline device.go]: Of course you can, just use a different interface:
```
// implements BootEnvReadWriter
type fakeBootEnv struct {
	readVars  BootVars
	readErr   error
	writeVars BootVars
	writeErr  error
}

func (f *fakeBootEnv) ReadEnv(...string) (BootVars, error) {
	return f.readVars, f.readErr
}

func (f *fakeBootEnv) WriteEnv(w BootVars) error {
	f.writeVars = w
	return f.writeErr
}

func Test_commitUpdate(t *testing.T) {
	device := device{}

	device.BootEnvReadWriter = &fakeBootEnv{
		readVars: BootVars{
			"upgrade_available": "1",
		},
	}

	if err := device.CommitUpdate(); err != nil {
		t.FailNow()
	}

	device.BootEnvReadWriter = &fakeBootEnv{
		readVars: BootVars{
			"upgrade_available": "1",
		},
		writeErr: errors.New("failed"),
	}
	if err := device.CommitUpdate(); err == nil {
		t.FailNow()
	}
}
```

Comment (pasinskim) [inline device.go]: @bboozzoo right, I haven't thought about that. Makes perfect sense. @oleorhagen I think what @bboozzoo is suggesting makes lot of sense.

Comment (oleorhagen) [inline device.go]: Yeh. Way prettier than our hack. I'll fix it up now.

Comment (oleorhagen) [inline device.go]: So I created a new branch, and a new pull request: #220

**Comment by kacf** [comment]
Superseded by #220.

**Pull request closed without merging**